### DISCUSSION TITLE
otlp_http.lib not built

### DISCUSSION THREAD

**Discussion by mititelud** [open-telemetry/opentelemetry-cpp#1565] [Q&A]
Hello,

I've been trying to build the otlp_http library in order to run the example_otlp_http example.
I've set up the dependecies needed and followed the opentelemetry build steps presented in repo_path/Install.md ### Building as standalone CMake Project
after i get to repo_path/build and run
cmake -DWITH_OTLP=ON -DWITH_OTLP_HTTP=ON ..  
or 
cmake ..
-- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.19044.
-- Building for architecture ARCH=x64
-- Found RE2 via CMake.
-- Could NOT find CURL (missing: CURL_LIBRARY) (found version "7.84.0")
-- PROTOBUF_PROTOC_EXECUTABLE=C:/work/vcpkg/installed/x64-windows/tools/protobuf/protoc.exe
-- gRPC_CPP_PLUGIN_EXECUTABLE=C:/work/vcpkg/installed/x64-windows/tools/grpc/grpc_cpp_plugin.exe
Using external nlohmann::json
GTEST_INCLUDE_DIRS   =
GTEST_BOTH_LIBRARIES = GTest::gtest;GTest::gtest_main
Building with nostd types...
-- Could NOT find CURL (missing: CURL_LIBRARY) (found version "7.84.0")
-- Could NOT find CURL (missing: CURL_LIBRARY) (found version "7.84.0")
PATH:C:/work/opentelemetry-cpp/examples/grpc/protos:C:/work/opentelemetry-cpp/examples/grpc/protos/messages.proto
-- Could NOT find CURL (missing: CURL_LIBRARY) (found version "7.84.0")
CMake Warning at examples/http/CMakeLists.txt:4 (message):
  Skipping http client/server example build: CURL not found

-- Configuring done
-- Generating done
-- Build files have been written to: C:/work/opentelemetry-cpp/build

I try to run the next step either with
cmake --build . --target install
or 
cmake --build . --target all_build
but the otlp_http library does not get built, while others are getting built (debug/opentelemetry_protol.lib, ... , examples/..../example_plugin.lib, exporters/...../opentelemetry_exporter_otlp_grpc, ... , sdk/..../opentelemetry_metrics.lib , sdk/..../opentelemetry_version.lib ).

Please help.

Thank you

**Comment by lalitb** [comment]
otlp_http target will not be built if CURL is not installed. Please install CURL and try again.

**Comment by mititelud** [comment]
thank you
it seems that the .lib was built using 
cmake .. -Wno-dev 
since other curl issues occured

**Comment by ThomsonTan** [comment]
Does `cmake .. -Wno-dev` fix the issue?

**Comment by mititelud** [comment]
Hello,

Yes, it fixed the issue, the opentelemetry_exporter_otlp_http library is now built.

**Comment by mititelud** [comment]
I am currently trying to build the example_otlp_http project, but this error occurs 
LNK1104	cannot open file 'CURL::libcurl-NOTFOUND.obj'	
I checked the project properties -> linker -> input and this is present CURL::libcurl-NOTFOUND as an additional dependency, I am not sure what this refers to.
If it removed, several unresolved external symbols errors occur because of the otlp_http_client_curl.lib
--------------------------------------one of the errors--------------------------------------------------------------------------
LNK2019	unresolved external symbol __imp_curl_multi_init referenced in function "public: __cdecl opentelemetry::v1::ext::http::client::curl::HttpClient::HttpClient(void)" (??0HttpClient@curl@client@http@ext@v1@opentelemetry@@QEAA@XZ)	example_otlp_http	C:\work\opentelemetry-cpp\build\examples\otlp\opentelemetry_http_client_curl.lib(http_client_curl.obj)	1	

If someone knows a fix for this issue, please let me know.

Thank you
